test/Test/Indigo/Contracts/HomebaseLite/PropertyTests.hs

```diff
@@ -114,17 +120,19 @@ hprop_main = withTests 500 $ property $ do
   testScenarioProps $ withInitialNow curTime $
     genericScenario tokenCount expectation \holder admin contract -> do
       maintainer <- newAddress "maintainer"
-      withSender admin do
-        call contract (Call @"Add_maintainers") [maintainer]
-      withSender maintainer do
-        call contract (Call @"Configure") conf
-      withSender holder do
-        call contract (Call @"Propose") (uri, #choices :! choices)
-      advanceTime $ sec (fromIntegralOverflowing @Natural delay)
-      for_ votes \vote -> do
-        voter <- newAddress auto
-        withSender voter do
-          call contract (Call @"Vote") (uri, #choice_index :! vote)
-            & if vote > maxChoice
-              then expectCustomErrorNoArg #noSuchChoice
-              else id
+      withImplicitAddress holder $ \iholder -> do
+        withImplicitAddress admin $ \iadmin -> do
```

Comment:
Wouldn't it be easier to change `genericScenario` to use `ImplicitAddress`es instead of converting those back and forth?